### ISSUE TITLE
Prepare for breaking API changes

### ISSUE THREAD

**Issue by NotJoeMartinez** [NotJoeMartinez/chess-insights#9]
*[TLDR](#tldr)*

Some time last month the chess.com developers announced they will be [requiring developer contact information](https://www.chess.com/announcements/view/breaking-change-user-agent-contact-info-required) in the headers of all requests to their public API. 

According to the post:
> Processes without a user-agent with contact information will return a 403 Forbidden response.

The deadline to integrate this into projects was June 20th (six days ago). I became aware of this yesterday. 
Fortunately this seems to have been temporarily delayed as all the requests made by chessinsights.xyz still work properly and I verified this with curl using deliberately empty headers. 

Their post doesn't directly address why they are rolling out authentication but reading through the lines on this sentence it seems they are experiencing an unusually high amount of traffic to their endpoints. 

> The user-agent should include a contact email and/or your chess.com username to allow communication about processes generating abnormal or excessive load on API endpoints

### 1. Add contact info user agent
Obviously this is the most important but there's a small risk of having requests with my user agent blacklisted by trusting random clients carry its reputation. In fact there's nothing stopping anyone from finding a developers email or chess.com username and going nuts with a python script. Maybe that's why this hasn't been rolled out yet? 

### 2. Temp storage of random users
Users spam requests for the same user within a short time frame. Cache the data in a database for a window of time and return it instead of triggering a new archive request every time. 

### 3. Persistent storage of high profile users
Permanently store an archive of celebrity accounts and regularly update them on a crontab to prevent 100 full downloads of GothamChess a week. 

### 4. oAuth for returning users
Requiring oAuth for API access seems to be where this is all going. Getting ahead of it now will help with more ambitious plans for this project that will require authentication anyway.  

## TLDR <a href="#tldr" id="tldr">#</a>
- The Chess.com API getting overloaded 
- All requests are now required to have contact info in headers
- The API is probably moving to authentication based access 
- My plans to adapt to these changes

**Comment by NotJoeMartinez** [comment]
This never happened
